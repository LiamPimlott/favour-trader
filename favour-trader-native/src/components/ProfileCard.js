import React, {Component} from 'react';
import {StyleSheet, View, Text, } from 'react-native';
import { Card, Avatar, Button, Icon} from 'react-native-elements'

class ProfileCard extends React.Component{

    render() {
        const { name } = this.props;
        const { email } = this.props;
        return (
            <View >
                <Card
                    title={name.first + ' ' + name.last}>
                    <View style={styles.avatar}>
                        <Avatar
                            large
                            rounded
                            source={{uri: "https://s3.amazonaws.com/uifaces/faces/twitter/ladylexy/128.jpg"}}
                            activeOpacity={0.7}
                        />
                    </View>

                    <Text style={styles.email} >
                        {email}
                    </Text>
                    
                    
                </Card>
                
            </View>
        )
    }
}

const styles = StyleSheet.create({
    email: {
        fontSize: 20,
        marginTop: 20,
        textAlign: 'center',
    },
    avatar: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        marginTop: 20,
        marginBottom: 20,
    },
    
});

export default ProfileCard;