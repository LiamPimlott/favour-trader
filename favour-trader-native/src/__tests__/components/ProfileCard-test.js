import 'react-native';
import React from 'react';
import { shallow, configure } from 'enzyme';
import moxios from "moxios";
import ProfileCard from '../../components/ProfileCard.js';
import toJson from 'enzyme-to-json'; //added this line

beforeEach(function () {
    const Adapter = require("enzyme-adapter-react-16");
    configure({adapter: new Adapter()});
    moxios.install();
});

describe('Testing ProfileCard component', () => {
    it('renders as expected', () => {
        const updateUser='';
        const isCurrentUser=false;
        const userProfile={
            firstName: '',
            lastName: '',
            country: '',
            state: '',
            city: '',
            postalCode: '',
            about: ''
        };
        const wrapper = shallow(
            <ProfileCard updateUser={updateUser} isCurrentUser={isCurrentUser} userProfile={userProfile}/>
        );
        expect(toJson(wrapper)).toMatchSnapshot(); //edited this line
    });
});