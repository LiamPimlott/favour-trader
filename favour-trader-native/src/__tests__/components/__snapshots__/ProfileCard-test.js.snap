// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ProfileCard component renders as expected 1`] = `
<View>
  <Card
    title=" "
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 20,
          "marginTop": 20,
        }
      }
    >
      <Avatar
        activeOpacity={0.7}
        editButton={
          Object {
            "iconColor": "#fff",
            "iconName": "mode-edit",
            "iconType": "material",
            "size": null,
            "style": null,
            "underlayColor": "#000",
          }
        }
        large={true}
        onEditPress={null}
        rounded={true}
        showEditButton={false}
        source={
          Object {
            "uri": "https://s3.amazonaws.com/uifaces/faces/twitter/ladylexy/128.jpg",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          City: 
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          State: 
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Country: 
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
          }
        }
      >
        <View />
      </View>
    </View>
  </Card>
  <View />
  <View />
</View>
`;
